workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/pages/TicketsOverview.tsx

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable quote-props */
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import {ClayInput} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import {ClayPaginationBarWithBasicItems} from '@clayui/pagination-bar';
+import {useState} from 'react';
+import {QueryClient, useMutation, useQueryClient} from 'react-query';
+
+import {RecentActivity} from '../components/RecentActivity';
+import {TicketGrid} from '../components/TicketGrid';
+import useDebounce from '../hooks/useDebounce';
+import {useRecentTickets} from '../hooks/useRecentTickets';
+import {useTickets} from '../hooks/useTickets';
+import {Liferay} from '../services/liferay';
+import {generateNewTicket} from '../services/tickets';
+import {Filter} from '../types';
+
+const DEBOUNCE_DELAY: number = 300;
+
+const INITIAL_FILTER_STATE: Filter = {
+	field: '',
+	label: '',
+	value: '',
+};
+
+const FILTERS: Filter[] = [
+	{
+		field: '',
+		label: 'No Filter',
+		value: '',
+	},
+	{
+		field: 'ticketStatus',
+		label: 'Open Issues',
+		value: 'open',
+	},
+	{
+		field: 'ticketStatus',
+		label: 'Queued Issues',
+		value: 'queued',
+	},
+	{
+		field: 'priority',
+		label: 'Major Priority Issues',
+		value: 'major',
+	},
+	{
+		field: 'resolution',
+		label: 'Unresolved Issues',
+		value: 'unresolved',
+	},
+];
+
+const TicketsOverview: React.FC = () => {
+	const queryClient: QueryClient = useQueryClient();
+
+	const [filter, setFilter] = useState(INITIAL_FILTER_STATE);
+	const [page, setPage] = useState(1);
+	const [pageSize, setPageSize] = useState(20);
+	const [search, setSearch] = useState<string>('');
+
+	const debouncedSearch = useDebounce(search, DEBOUNCE_DELAY);
+	const debouncedPage = useDebounce(page, DEBOUNCE_DELAY);
```

Comment:
FYI, debouncing both page and search separately will still result in them getting changed at slightly different times. This isn't ideal, it would be better to have a debounced function that sets page and search at the same time. I don't think this is going to cause any crazy issues though, so you don't need to change it, but I just wanted you to be aware that there is a better approach.

Reply:
Thanks @ethib137 , I thought also the same!